### PR TITLE
Don't mark builder status unhealthy if mev-boost status is non 200

### PR DESCRIPTION
**Pull request by terencechain**
**What type of PR is this?**

> Bug fix

**What does this PR do? Why is it needed?**

If mev-boost or relayers are unavailable, Prysm's builder service's health endpoint will report unhealthy due to builder APIs status check. Depending on the usage,  users can mark the beacon node as unavailable instead of defaulting to local execution engine block production.
This PR added a ticker to spawn off during `Start` to ping relayer status every min and log the error if the relayer endpoint returns non-200 

**Which issues(s) does this PR fix?**

Fixes #

**Other notes for review**

The fundamental limitation is health status is binary. There's no gradient to report degraded status. This can be improved upon in another PR(s)

### CODE REVIEW

File: beacon-chain/builder/service.go [hunk 1]
```diff
@@ -117,7 +117,11 @@ func (s *Service) Status() error {
 		return nil
 	}
 
-	return s.c.Status(ctx)
+	if err := s.c.Status(ctx); err != nil {
+		log.WithError(err).Error("Error calling status endpoint, are mev-boost or relayers down?")
+	}
```
[the remark this answers is not in the document]
Comment (terencechain) [reply beacon-chain/builder/service.go:122]: > Is there a more appropriate place to log this? 
We only check builder status here and once during start-up. You are right that it'll be easily missed, but I thought it was better than nothing. Another thing we can do is to remove the calling builder status here. I'm not entirely sure what is best here

[the remark this answers is not in the document]
Comment (terencechain) [reply beacon-chain/builder/service.go:122]: Another thought is to implement a ticker that pulls relayer status every min for logging

File: beacon-chain/builder/service_test.go [hunk 2]
```diff
@@ -37,3 +38,12 @@ func Test_RegisterValidator(t *testing.T) {
 	require.NoError(t, s.RegisterValidator(ctx, []*eth.SignedValidatorRegistrationV1{{Message: &eth.ValidatorRegistrationV1{Pubkey: pubkey[:], FeeRecipient: feeRecipient[:]}}}))
 	assert.Equal(t, true, builder.RegisteredVals[pubkey])
 }
+
+func TestService_PollRelayerStatus_ContextCanceled(t *testing.T) {
```
Comment (rauljordan) [inline beacon-chain/builder/service_test.go:42]: This doesn't test anything, better to just remove

Comment (terencechain) [reply beacon-chain/builder/service_test.go]: You are right. I thought it would be nice for code coverage regardless. Removed :)

**Review by rauljordan — DISMISSED** [review]

**Review by james-prysm — APPROVED** [review]

**Pull request merged**